File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.java
class Solution {
    public void setZeroes(int[][] matrix) {
        ArrayList<ArrayList<Integer>> list = new ArrayList<>();
        int n = matrix.length;
        int m = matrix[0].length;
        for(int i=0;i<n;i++){
            for(int j=0;j<m;j++){
                if(matrix[i][j] == 0){
                    ArrayList<Integer> pair = new ArrayList<>();
                    pair.add(i);
                    pair.add(j);
                    list.add(pair);
                }
            }
        }
        for(ArrayList<Integer> pair : list){
            int a = pair.get(0);
            int b = pair.get(1);
            for(int i=0;i<m;i++){
                matrix[a][i] = 0;
            }
            for(int j=0;j<n;j++){
                matrix[j][b] = 0;
            }
        }
    }
}

File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.cpp
class Solution {
public:
    void setZeroes(vector<vector<int>>& matrix) {
        int m = matrix.size();
        int n = matrix[0].size();
        vector<pair<int, int>> v;
        for(int i=0;i<m;i++){
            for(int j=0;j<n;j++) {
                if(matrix[i][j] == 0) v.push_back({i, j});
            }
        }
        for(int i=0;i<v.size();i++) {
            // convert row to 0
            int x = v[i].first;
            for(int j=0;j<n;j++){
                matrix[x][j] = 0;
            }

            // convert col to 0
            int y = v[i].second;
            for(int j=0;j<m;j++){
                matrix[j][y] = 0;
            }
        }
    }
};